### PR TITLE
Lots of dependency updates

### PR DESCRIPTION
**Pull request by RDIL**
- Yarn v3 (because its cool)
- Update TypeScript to v4
- Update Relay to v12
- Tried to update graphql to v16 but that will have to be for another day
- Updated react-scripts to v4 because whatever version it was on was really old
- ESLint got updated in the process, which includes the rule `eslint-plugin-import/no-anonymous-default-exports`

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -7,9 +7,8 @@ env:
 
 task:
   node_modules_cache:
-    folder: node_modules
-    fingerprint_script: cat yarn.lock
-    populate_script: yarn
+    folder: .yarn/cache
+  install_script: yarn
```
Comment (RDIL) [inline .cirrus.yml:11]: This change has been made because Yarn 3 downloads package tarballs to `.yarn/cache` and then extracts them to `node_modules`, however n_m is platform dependent, so it's not really safe to cache it.  Instead, this just downloads the current cache, and runs `yarn` to download any new modules, unpack them, and as an added bonus, automatically remove any unused packages, so no need to re-populate the cache completely just because a dependency has changed (just let yarn modify the parts that have changed).

TL;DR: This change makes it so that yarn can properly manage the packages, and it will ensure that it cleans up after itself.

File: package.json [hunk 2]
```diff
@@ -85,6 +85,12 @@
     "sync-schema": "get-graphql-schema https://api.cirrus-ci.com/schema.json | sed -e 's/Long/Int/g' > schema.graphql && prettier --write schema.graphql",
     "bootstrap": "yarn sync-schema && yarn relay"
   },
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
```
Comment (RDIL) [inline package.json:93]: This is expected to be present now.

**Comment by RDIL** [comment]
@fkorotkov PTAL

File: src/components/common/CirrusFavicon.tsx [hunk 3]
```diff
@@ -53,7 +53,7 @@ interface CirrusFaviconProps {
   status?: BuildStatus | TaskStatus | boolean;
 }
 
-export default (props: CirrusFaviconProps) => {
+export default function CirrusFavicon(props: CirrusFaviconProps): JSX.Element {
```
Comment (fkorotkov) [inline src/components/common/CirrusFavicon.tsx:56]: Was there an error here? Or is it some kind of automatic fix?

**Review by fkorotkov — COMMENTED** [review]
Thank you! Have a couple of questions.

Are files in ` .yarn/plugins` and ` .yarn/releases` are necessary to commit?

File: src/components/common/CirrusFavicon.tsx (hunk 3 above)
Comment (RDIL) [reply src/components/common/CirrusFavicon.tsx:56]: This PR updated eslint, which introduced a new rule that suggests against using anonymous default exports, so I used next.js's codemod to fix it.

As for .yarn/plugins and .yarn/releases, the idea behind it is that every repository keeps it's own copy of the package manager.  It's like the gradle wrapper, but instead of a wrapper it's the whole thing.  It has to be committed to work.

Comment (fkorotkov) [reply src/components/common/CirrusFavicon.tsx:56]: What do you think of using [non zero install](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored). We do still download things from the internet so I guess there is not that much benefit of commiting these files and keeping them up to date.

Comment (RDIL) [reply src/components/common/CirrusFavicon.tsx:56]: I've had long conversations about zero installs in the yarn discord, and I ultimately think using zero-installs is a bad idea.

**Review by fkorotkov — APPROVED** [review]
Thank you!
